packages/mobile/src/navigator/NavigationService.ts

```diff
@@ -148,13 +150,15 @@ export async function ensurePincode(): Promise<boolean> {
 
   try {
     await requestPincodeInput(true, false)
+    ValoraAnalytics.track(AuthenticationEvents.get_pincode_complete)
     return true
   } catch (error) {
     if (error === CANCELLED_PIN_INPUT) {
       Logger.warn(`${TAG}@ensurePincode`, `PIN entering cancelled`, error)
     } else {
       Logger.error(`${TAG}@ensurePincode`, `PIN entering error`, error)
     }
+    ValoraAnalytics.track(AuthenticationEvents.get_pincode_error)
```

Comment:
If the user cancelled do we still count it as an error? Maybe we do (ie they don't remember their PIN so they cancel), just asking

Reply:
yeah good question! i think we do want this error regardless if it is a user action or genuine failure, because if these events are meant to help us understand the success rate of the pin flow with and without the biometrics, then we need to capture the user cancelled/forgot their pin error and count that towards the failure rate. but yeah i'm only 95% convinced about this, i'd like to leave it like this and ask gowtham to have a play with the nightly build tomorrw to see if the events are what he expects. and i can tweak if they are not :)